### PR TITLE
Show Select All button only to logged users

### LINKED ISSUE #2874: Flow breech?
On public libraries, I'm **NOT** logged in and it allows me to _''Select All'_' entities, and shows the bulk publishing and editing option. 

It allows to go through all the steps and when you hit "save" you are asked to log-in. 
Nevertheless, this is a major flow issue. It should not be accessible for **non-logged** users.

### PR DESCRIPTION
**Pull request by fnocetti**
fixes #2874 

PR checklist:
- [ ] Update READ.me ?
- [ ] Update API documentation ?

QA checklist:
- [ ] Smoke test the functionality described in the issue
- [ ] Test for side effects
- [ ] UI responsiveness
- [ ] Cross browser testing
- [ ] Code review

### CODE REVIEW

**Comment by RafaPolit** [comment]
Please try to refactor the file that went beyond the 250 lines.  It's not overly critical, but it would be nice given this is a small development to go in and improve that file a bit.  Let me know if there is chance, otherwise I will merge this as is.

**Comment by fnocetti** [comment]
I'm removing the Ready for QA tag while I increase the coverage.

File: app/react/Layout/specs/DocumentsList.spec.js [hunk 1]
```diff
@@ -90,6 +91,16 @@ describe('DocumentsList', () => {
...
+      props.connections = { totalRows: 2 };
+      props.connectionsGroups = Immutable.fromJS([
+        Immutable.fromJS({
+          templates: Immutable.fromJS([
+            {
+              count: Immutable.fromJS(2),
+            },
+          ]),
+        }),
+      ]);
```
Comment (RafaPolit) [inline app/react/Layout/specs/DocumentsList.spec.js:103]: Immutable already works recursively.  `Immutable.fromJS([{ a: 'b' }])` will create an instance of Immutable.Map on that first element.  There is no need to create immutables from the children.  Even more so with numeric values... you do not need to make `2` an immutable!

**Comment by RafaPolit** [comment]
There is a very minor change needed, please fix that and then rebase or merge dev into this branch.  Lets hope tests pass with the same luck you have had up until now! :)

File: app/react/Layout/specs/DocumentsList.spec.js (hunk 1 above)
Comment (fnocetti) [reply app/react/Layout/specs/DocumentsList.spec.js:103]: That's an ugly one. Getting to it!

**Comment by RafaPolit** [comment]
LGTM

**Pull request merged**